sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/DynamicJoinSelection.scala

```diff
@@ -50,10 +52,41 @@ object DynamicJoinSelection extends Rule[LogicalPlan] {
       mapStats.bytesByPartitionId.forall(_ <= maxShuffledHashJoinLocalMapThreshold)
   }
 
-  private def selectJoinStrategy(plan: LogicalPlan): Option[JoinStrategyHint] = plan match {
+  private def selectJoinStrategy(
+      join: Join,
+      plan: LogicalPlan): Option[JoinStrategyHint] = plan match {
     case LogicalQueryStage(_, stage: ShuffleQueryStageExec) if stage.isMaterialized
       && stage.mapStats.isDefined =>
-      val demoteBroadcastHash = shouldDemoteBroadcastHashJoin(stage.mapStats.get)
+
+      val manyEmptyInPlan = shouldDemoteBroadcastHashJoin(stage.mapStats.get)
+      val canBroadcastPlan = (canBuildBroadcastLeft(join.joinType) && join.left == plan) ||
```

Comment:
instead of checking `join.left == plan` here, I think this method `selectJoinStrategy` should simply take a `isLeft: Boolean` parameter instead of `plan: LogicalPlan`